### ISSUE TITLE
raspberry pi 4 + buster - h265 possible?

### ISSUE THREAD

**Issue by danpowpow** [moonlight-stream/moonlight-embedded#760]
before i go crazy troubleshooting an impossible situation:

according to the link below, the pi4 supports h265 decode
https://cdn.shopify.com/s/files/1/1560/1473/files/Inside_Raspberry_Pi_4.pdf?2142

but when i run moonlight with -codec h265 i get:
Initializing platform...done
Resolving host name...done
Starting RTSP handshake...done
Initializing control stream...done
Initializing video stream...done
Initializing audio stream...done
Initializing input stream...done
Starting control stream...done
Video format not supported
Starting video stream...Video stream start failed: -1

Has anyone succeeded with moonlight embedded and pi4 / h265?
cheers

**Comment by yukatan** [comment]
Same problem here. I guess the OS aren't ready to use this codec on the raspberry yet.

**Comment by discordianfish** [comment]
Trying to do the same but I don't think it's possible :(. At least platform=pi only support h264: https://github.com/irtimmer/moonlight-embedded/blob/7a32234c0b2fd7603a2fe93640209c3a5bf98495/src/video/pi.c#L54
Same applies to moonlight-pc where general rpi hw support was added, but not for h265,

And since that's the only way to do 4K streaming, it doesn't look like that's possible. Same btw seems to apply to parsec and steamlink.

**Comment by yukatan** [comment]
raspberry pi 4 has the hardware to decode h265 but software is not ready yet. it won't be ready until Q1-Q2 2020 sadly

**Comment by discordianfish** [comment]
@yukatan libelec supports h265 hw decoding on rpi though: https://libreelec.tv/2019/06/libreelec-9-2-alpha1-rpi4b/
So it seems possible, here are a few more details: https://forum.kodi.tv/showthread.php?tid=345112

Also see the same discussion over at moonlight-qt: https://github.com/moonlight-stream/moonlight-qt/issues/160

**Comment by zaptrem** [comment]
Any change on this?

**Comment by seeul8er** [comment]
Is this desirable for real-time video streaming? While you can reduce the bitrate using H265 over H264 you will however induce more latency since encoding and decoding the newer standards takes considerably longer. I'd go test the encoding/decoding speed first before spending too much time on the implementation.

**Comment by zbuc** [comment]
I am experiencing this issue and willing to help implement a patch if I can figure it out.

Has anyone else made any progress on this?

**Comment by zbuc** [comment]
Is this project dead?

**Comment by zaptrem** [comment]
No, but one of the other Moonlight versions works natively on RPi now. It’s
much better.

**Comment by KiralyCraft** [comment]
Yes this is now possible on Moonlight-Qt version 3.1.2
